package.json

```diff
@@ -7,20 +7,38 @@
     "@types/node": "^8.0.16",
     "@types/react": "^15.0.39",
     "@types/react-dom": "^15.5.1",
+    "eslint-config-prettier": "^2.3.0",
+    "husky": "^0.14.3",
+    "lint-staged": "^4.0.2",
+    "prettier": "^1.5.3",
+    "prettier-eslint": "^6.4.2",
     "react": "^15.6.1",
     "react-dom": "^15.6.1",
     "react-scripts-ts": "2.4.0"
   },
   "devDependencies": {
     "@types/enzyme": "^2.8.4",
     "enzyme": "^2.9.1",
+    "react-addons-test-utils": "^15.6.0",
+    "tslint-config-prettier": "^1.1.0",
     "react-test-renderer": "^15.6.1",
     "tslint-eslint-rules": "^4.1.1"
   },
   "scripts": {
     "start": "react-scripts-ts start",
     "build": "react-scripts-ts build",
     "test": "react-scripts-ts test --env=jsdom",
-    "eject": "react-scripts-ts eject"
+    "eject": "react-scripts-ts eject",
+    "pre-commit": "lint-staged"
+  },
+  "lint-staged": {
```

Comment:
So, lint-staged means that all of our files which are committed via git are formatted by prettier. Advantageous, because now you don't have to worry about formatting while you are writing code, plus if we have new people/ other people commit, prettier is going to format their code for us.  

If you want to format as you are writing code, you can add prettier to your IDE (I did it to mine). If you want more info on this go here:

https://github.com/prettier/prettier#pre-commit-hook

https://github.com/prettier/prettier#editor-integration